### ISSUE TITLE
[BUG] 'Index Queue Initialization' runs into an error in TypoScriptConfiguration::getInitialPagesAdditionalWhereClause()

### ISSUE THREAD

**Issue by ralfhueskes** [TYPO3-Solr/ext-solr#3121]
**Describe the bug**
After I try to create the queue in the module "Index Queue" via "Index Queue Initialization" (pages, news) I get the following error message:

**Argument 1 passed to ApacheSolrForTypo3\Solr\System\Configuration\TypoScriptConfiguration::getInitialPagesAdditionalWhereClause() must be of the type string, null given, called in /html/provendis.typo3cms.11/private/typo3conf/ext/solr/Classes/Domain/Site/Site.php on line 180**

I tried it with my own Typoscript Config, as well as with the defaulr Config in the SOLR Extension.

**To Reproduce**
1. Go to Modul 'Index Queue'
2. Click on 'Clear Index Queue'
3. Check on 'pages, news, etc' at Index Queue Initialization -> Click on 'Queue Selected Content for indexing'
4. See the error above

**Expected behavior**
No error ;) 

**Screenshots**
Upload not possible here?

**Used versions (please complete the following information):**
 - TYPO3 Version: 11.5.4-dev
 - Browser: chrome, ff
 - EXT:solr Version: dev-task/2976_TYPO3.11_compatibility
 - Used Apache Solr Version: Solr 7.6.0
 - PHP Version: 7.4.2

**Additional context**

**Comment by dkd-kaehm** [comment]
@ralfhueskes 
I can't reproduce that issue.
Are there mount points? Custom page types?
Something else, which is not default/standard?
Does it happen on initializing pages queue only? Or news only?

**Comment by ralfhueskes** [comment]
@dkd-kaehm 
There are only Standard pages, some menu separators and shortcuts. 
I get the error with 'pages' or 'news'. Basically even if I have only included 'pages' via the SOLR EXT configuration setup.

I have tested it with different SOLR server. 
DDEV, Mittwald, but always the same behavior. 

Only VHS, NEWS is in use at external extensions. 
Basic Config via Sitepackage, as recommended

**Comment by dkd-kaehm** [comment]
@ralfhueskes thanks for reporting and responses.
I'll try to hardening the method and its neighborhood and try to reproduce.

**Comment by dkd-kaehm** [comment]
@ralfhueskes

**Comment by dkd-kaehm** [comment]
@ralfhueskes 
Please see in TypoScript object browser the value of `plugin.tx_solr.index.queue.pages` or other queue config for `pages` table.
Is it 0?
![EXT_solr_3121_queue_init_exception](https://user-images.githubusercontent.com/25453927/145597002-637ead19-3a3a-4706-9c25-85cc4c82f284.jpg)

Seems to be related to #2742

**Comment by ralfhueskes** [comment]
@dkd-kaehm:
all configs have value = 1

For example 'pages', looks good for me
```
[plugin][index][queue]

[pages] = 1 
[initialization] = ApacheSolrForTypo3\Solr\IndexQueue\Initializer\Page
[allowedPageTypes] = 1,4,7 
[indexingPriority] = 0
[indexer] = ApacheSolrForTypo3\Solr\IndexQueue\PageIndexer
[additionalWhereClause] = (doktype = 1 OR doktype=4 OR (doktype=7 AND mount_pid_ol=0)) AND no_search = 0 
[excludeContentByClass] = typo3-search-exclude
```

**Comment by dkd-kaehm** [comment]
@ralfhueskes 
Can you please debug and find out the cause, why the `$solrConfiguration` is not fully initialized.

https://github.com/TYPO3-Solr/ext-solr/blob/6023d78d761a9618eb9f66868b10c4b09f02946c/Classes/Domain/Site/Site.php#L178-L180

Also why returns the following method the `null` for your sites root page uid in your instance?:

https://github.com/TYPO3-Solr/ext-solr/blob/6c88011540c0f214afda52a34c398a5880231c09/Classes/FrontendEnvironment/Tsfe.php#L189-L193

---

Apart from this: Is all the stuff green in Reports module on EXT:solr section?

**Comment by dkd-kaehm** [comment]
@ralfhueskes Is the site in composer mode?

**Comment by ralfhueskes** [comment]
@dkd-kaehm Yes, its composer and all reports are green!

i have run some debugs in the above files and nothing seems unusual. 

However, by accident I changed the typoscript value for 'additionalWhereClause' and set it to
`additionalWhereClause = doktype IN(1, 7)`  as a test. Then the error no longer appears and 
the indexing works perfectly. 

`additionalWhereClause = (doktype = 1 OR doktype=4 OR (doktype=7 AND mount_pid_ol=0)) AND no_search = 0`
throws me the error again. 

For me as a small workaround i just override the default setup.
I'll take another look at why the other string doesn't work.

**Comment by dkd-kaehm** [comment]
That is really strange.

The error happens a little bit before the TypoScript value of `additionalWhereClause` should be fetched.

The error message

```
Argument 1 passed to ApacheSolrForTypo3\Solr\System\Configuration\TypoScriptConfiguration::getInitialPagesAdditionalWhereClause() must be of the type string, null given, called in /html/provendis.typo3cms.11/private/typo3conf/ext/solr/Classes/Domain/Site/Site.php on line 180
```
says, that the var `$indexQueueConfigurationName` has the value `=null` and that because the var `$solrConfiguration` does not have any indexing configuration for `pages` table.

 https://github.com/TYPO3-Solr/ext-solr/blob/6023d78d761a9618eb9f66868b10c4b09f02946c/Classes/Domain/Site/Site.php#L178-L180

**Note:** EXT:solr respects the TypoScript configurations assigned to Sites root page. All other includes in sub tree of site are ignored. @ralfhueskes Please make sure the EXT:solr base template is included in sites root page and nowhere else.

---
#### Solution way.

Nevertheless I'll remove the `additionalWhereClause` awareness in `getPagesWithinNoSearchSubEntriesPages()` methods, 
because nothing except the pages within `no_search_sub_entries` matters to exclude them from indexing. 
Also we do not want to care about any "where clauses" or some other configurations if we fetching the pages within `no_search_sub_entries`, to prevent re-including some pages within that list.
Moreover I'll deprecate the method `System\Records\Pages\PagesRepository::findAllPagesWithinNoSearchSubEntriesMarkedPagesByRootPage(int $rootPageId)` and replace it with 
`System\Records\Pages\PagesRepository::findAllPagesWithinNoSearchSubEntriesMarkedPages()` which will not matter about the site trees and fetch simply all those pages in whole instance. Those pages will be used as filter (PIDs without PIDs-WithinNoSearchSubEntries) on all places.

So following methods will be modified:

https://github.com/TYPO3-Solr/ext-solr/blob/6023d78d761a9618eb9f66868b10c4b09f02946c/Classes/Domain/Site/Site.php#L165-L187

https://github.com/TYPO3-Solr/ext-solr/blob/a8f1115921834d0c5bde86d98ad6f5e69f504700/Classes/System/Records/Pages/PagesRepository.php#L208-L242

**Comment by ralfhueskes** [comment]
@dkd-kaehm Thanks for your execution and upcoming adjustments. To clarify: I only include the SOLR Typoscript configuration through my site package. 
If possible I will test again with the default whereClause value. At least we could now identify the cause at this point!

**Comment by dkd-kaehm** [comment]
@ralfhueskes 
I can't still not reproduce it in my systems, could you please check the suggestion from #3130

**Comment by Apen** [comment]
I have this bug too and the double typoscript "start()" thing is not the problem here. 
I don't have dig more on this but change to `additionalWhereClause = doktype IN(1, 7)` works for me too.

**Comment by dkd-kaehm** [comment]
@Apen and @ralfhueskes 
It would be super if you could try to reproduce the both troubles in https://github.com/TYPO3-Solr/solr-ddev-site/tree/task/2976_TYPO3.11_compatibility

**Comment by dkd-kaehm** [comment]
@ralfhueskes 
Please try current state of [task/2976_TYPO3.11_compatibility](https://github.com/TYPO3-Solr/ext-solr/tree/task/2976_TYPO3.11_compatibility) branch.

**Comment by ralfhueskes** [comment]
@dkd-kaehm @Apen, 
I installed the update and the change to [additionalWhereClause] = (doktype = 1 OR doktype=4 OR (doktype=7 AND mount_pid_ol=0)) AND no_search = 0 worked without error now. The changes seem to work fine here!! Great work.

**Comment by ralfhueskes** [comment]
> @Apen and @ralfhueskes It would be super if you could try to reproduce the both troubles in https://github.com/TYPO3-Solr/solr-ddev-site/tree/task/2976_TYPO3.11_compatibility

This works absolutly perfect at my local machine. I could not reconstruct the error in TYPO3 v10 before either.

**Comment by Apen** [comment]
works for me too, good job

**Comment by dkd-kaehm** [comment]
Fixed in #3136
